GenderPayGap.WebUI/Controllers/Admin/AdminDataMigrationController.cs

```diff
@@ -0,0 +1,648 @@
+        {
+            if (!IsUserAllowedToImportData())
+            {
+                return Unauthorized();
+            }
+
+            viewModel.ParseAndValidateParameters(Request, m => m.Hostname);
+            viewModel.ParseAndValidateParameters(Request, m => m.Password);
+
+            if (string.IsNullOrWhiteSpace(Global.DataMigrationPassword))
+            {
+                // Don't allow access if no password is set
+                return Unauthorized();
+            }
+            if (viewModel.Password != Global.DataMigrationPassword)
+            {
+                // Don't allow access is the user got the password wrong
+                return Unauthorized();
+            }
+
+            StartChunkedResponse();
+
+            WriteParagraph($"Requesting data export from {viewModel.Hostname}");
+
+            string requestUrl = $"https://{viewModel.Hostname}/admin/data-migration/export-all?password={viewModel.Password}";
+            string responseString = new HttpClient().GetStringAsync(requestUrl).Result;
+
+            WriteParagraph($"Received data export from {viewModel.Hostname}");
+
+            ImportDataFromJsonString(responseString);
+
+            WriteParagraph($"Data Migration Complete!");
+
+            EndChunkedResponse();
+            return null;
+        }
+
+        private bool IsUserAllowedToImportData()
+        {
+            bool databaseIsEmpty = dataRepository.GetAll<User>().Count() == 0;
+
+            bool userIsAdministrator = User.Identity.IsAuthenticated
+                                       && ControllerHelper.GetGpgUserFromAspNetUser(User, dataRepository).IsAdministrator();
+
+            return databaseIsEmpty || userIsAdministrator;
+        }
+
+        private void StartChunkedResponse()
+        {
+            HttpContext.Items.Add("StartTime", DateTime.Now);
+
+            Response.StatusCode = 200;
+            Response.ContentType = "text/html";
+            Response.StartAsync().Wait();
+
+            Response.BodyWriter.WriteAsync(Encoding.UTF8.GetBytes(AdminMigrationTopAndBottomOfView.TopOfView));
+        }
+
+        private void WriteParagraph(string text)
+        {
+            DateTime startTime = (DateTime) HttpContext.Items["StartTime"];
+            TimeSpan elapsedTime = DateTime.Now.Subtract(startTime);
+
+            string html = $@"<p class=""govuk-body"">{elapsedTime.ToString("mm\\:ss")} {text}</p>";
+
+            Response.BodyWriter.WriteAsync(Encoding.UTF8.GetBytes(html));
+            Response.BodyWriter.FlushAsync();
+        }
+
+        private void EndChunkedResponse()
+        {
+            Response.BodyWriter.WriteAsync(Encoding.UTF8.GetBytes(AdminMigrationTopAndBottomOfView.BottomOfView));
+            Response.CompleteAsync().Wait();
+        }
+
+        private void ImportDataFromJsonString(string fileAsString)
+        {
+            AllData allData = JsonConvert.DeserializeObject<AllData>(fileAsString);
+
+            string deleteAllSql = @"
```

Comment:
I reckon we can delete all records of a table by "DELETE FROM TABLE" without any where clause

Reply:
I tried to do `DELETE FROM Table` in the UI (DataGrip) but got an error `Unsafe query: 'Delete' statement without 'where' clears all data in the table`.
Maybe this is just in the UI though?

Reply:
I'm fairly sure it's just the UI